benchmarks/battleship/benchmark/src/main/java/com/android/designcompose/benchmark/battleship/benchmark/ClickPlayfield.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose.benchmark.battleship.benchmark
+
+import androidx.benchmark.macro.ExperimentalMetricApi
+import androidx.benchmark.macro.StartupMode
+import androidx.benchmark.macro.TraceSectionMetric
+import androidx.benchmark.macro.TraceSectionMetric.Mode
+import androidx.benchmark.macro.junit4.MacrobenchmarkRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Until
+import com.android.designcompose.DCTraces
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * This is a rough example of a benchmark, though one that doesn't focus on any behavior well.
+ * Essentially it benchmarks a large recomposition, by clicking the Opponent's Playfield button,
+ * which causes the full screen to re-draw.
+ */
+@OptIn(ExperimentalMetricApi::class)
+@RunWith(AndroidJUnit4::class)
+class ClickPlayfield {
+
+    @get:Rule val benchmarkRule = MacrobenchmarkRule()
+
+    @Test
+    fun changePlayers() =
+        benchmarkRule.measureRepeated(
+            packageName = PACKAGE_NAME,
+            metrics =
+                listOf(
+                    TraceSectionMetric(DCTraces.DESIGNDOCINTERNAL),
+                    TraceSectionMetric(DCTraces.DESIGNVIEW),
+                    TraceSectionMetric(DCTraces.DESIGNFRAME_DE_SUBSCRIBE, Mode.Sum),
+                    TraceSectionMetric(DCTraces.DESIGNFRAME_FINISHLAYOUT, Mode.Sum),
+                    TraceSectionMetric(DCTraces.DESIGNTEXT_DE, Mode.Sum),
+                    TraceSectionMetric(DCTraces.JNIADDNODES, Mode.Sum),
+                    TraceSectionMetric(DCTraces.DOCSERVER_DOC, Mode.Sum),
+                    TraceSectionMetric(DCTraces.DESIGNVIEW_INTERACTIONSCOPE, Mode.Sum),
+                ),
+            iterations = 5,
+            /**
```

Comment:
This is because (as confusing as this is) COLD benchmarks kill the process between the setupBlock and the measureBlock. So if you have a cold benchmark you need to start the activity in the measure block.

Apparently this was done to match the behaviour of hawkeye, but it causes constant confusion

Reply:
Ok, I think I get it. Is this a valid way to reset the app's state between runs? I don't want to capture startup behavior in the benchmark, I specifically want to test a specific interaction. I could always write code to click the right buttons on the app to get it back to home, but then I'd be worried about the result being affected by some state that was left behind, and not a relatively clean state. 

Like, it makes sense to try to measure the first interaction with an app, right?

Reply:
Yeah what you are doing is valid. I think you can actually pass null to startup mode and then completely manage it yourself

Reply:
Yep, that works, and makes it so that it doesn't start up the app for the warm start only for me to kill it on the first go.